src/dfi/consensus/tokens.cpp

```diff
@@ -90,16 +90,22 @@ Res CTokensConsensus::operator()(const CCreateTokenMessage &obj) const {
         return res;
     }
 
+    const auto &consensus = txCtx.GetConsensus();
+    const auto height = txCtx.GetHeight();
+    const auto &tx = txCtx.GetTransaction();
+    auto &mnview = blockCtx.GetView();
+
     CTokenImplementation token;
     static_cast<CToken &>(token) = obj;
 
     auto tokenSymbol = trim_ws(token.symbol).substr(0, CToken::MAX_TOKEN_SYMBOL_LENGTH);
     auto tokenName = trim_ws(token.name).substr(0, CToken::MAX_TOKEN_NAME_LENGTH);
-
-    const auto &consensus = txCtx.GetConsensus();
-    const auto height = txCtx.GetHeight();
-    const auto &tx = txCtx.GetTransaction();
-    auto &mnview = blockCtx.GetView();
+    if (height >= consensus.DF22MetachainHeight) {
+        tokenName = (tokenName).substr(0, CToken::MAX_DST20_TOKEN_NAME_LENGTH);
```

Comment:
Wouldn't it be better to fail here rather than trim the `tokenName` and have unexpected results ?

Reply:
Definitely agree, this would be way better and more predictive behaviour.

Reply:
Resolved - added token name byte size limit check in e32f472